### PR TITLE
Docs: add `kwargs` type to fix formatting

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

As the title indicates: in several places our docs `kwargs` did not include its type, which made our doc builder treat it like a continuation of the previous parameter 💔 

Example of currently broken docs ([this function](https://huggingface.co/docs/transformers/v4.30.0/en/main_classes/processors#transformers.ProcessorMixin.save_pretrained)):
Screenshot 2023-07-10 at 18 20 47

This PR is a result of CMD+F on the broken pattern, and applying the fix :)

### CODE REVIEW

File: src/transformers/models/jukebox/tokenization_jukebox.py [hunk 1]
```diff
@@ -202,9 +202,6 @@ def prepare_for_tokenization(
         """
         Performs any necessary transformations before tokenization.
 
-        This method should pop the arguments from kwargs and return the remaining `kwargs` as well. We test the
```
Comment (gante) [inline src/transformers/models/jukebox/tokenization_jukebox.py:205]: This method doesn't have `kwargs` 🤔

**Comment by gante** [comment]
After the PR (as seen in the doc preview):
Screenshot 2023-07-10 at 18 55 21

**Review by amyeroberts — COMMENTED** [review]
Thanks for the work improving our docstrings! 

I'm more of the opinion that the doc builder should be updated (if possible) so that kwargs don't need type information. Having `Dict[str, Any]` seems a bit redundant. Let's get @sgugger's opinion as he knows the doc string patterns and decisions better than me :)

**Review by sgugger — APPROVED** [review]
Thanks for the fixes!

**Comment by gante** [comment]
@amyeroberts if you don't oppose, I'll merge this PR 🤗

**Comment by amyeroberts** [comment]
@gante Go for it!

**Pull request merged**